src/macaron/build_spec_generator/cli_command_parser/gradle_cli_command.py

```diff
@@ -0,0 +1,388 @@
+            configuration_cache_problems=parsed_arg.configuration_cache_problems,
+            gradle_user_home=parsed_arg.gradle_user_home,
+            init_script=parsed_arg.init_script,
+            include_build=parsed_arg.include_build,
+            write_verification_metadata=parsed_arg.write_verification_metadata,
+            max_workers=parsed_arg.max_workers,
+            project_dir=parsed_arg.project_dir,
+            priority=parsed_arg.priority,
+            project_cache_dir=parsed_arg.project_cache_dir,
+            update_locks=parsed_arg.update_locks,
+            warning_mode=parsed_arg.warning_mode,
+            exclude_task=parsed_arg.exclude_task,
+            system_prop=GradleCLIOptions.parse_properties(parsed_arg.system_prop) if parsed_arg.system_prop else None,
+            project_prop=(
+                GradleCLIOptions.parse_properties(parsed_arg.project_prop) if parsed_arg.project_prop else None
+            ),
+            tasks=parsed_arg.tasks,
+        )
+
+    @staticmethod
+    def parse_properties(props: list[str]) -> dict[str, str]:
+        """Return a dictionary that maps between a property and its value.
+
+        Each property definition value in `props` can have either of these format:
+        - `property=value` (e.g. `property=value` from `-Dproperty=value`): this will
+        be parsed into a dictionary mapping of `"property": "value"`.
+        Both the key and value of this mapping is of type string.
+        - `property` (e.g. `property` from `-Dproperty`): this will be parsed into a
+        dictionary mapping of `"property": <empty_string>`.
+
+        Parameters
+        ----------
+        props: list[str]
+            The list of properties definition provided in the cli command.
+            This is the list parsed by argparse.
+
+        Returns
+        -------
+        dict[str, str]:
+            The properties dictionary.
+
+        Examples
+        --------
+        >>> GradleCLIOptions.parse_properties(["boo=true", "foo=1", "bar"])
+        {'boo': 'true', 'foo': '1', 'bar': ''}
+        """
+        system_props = {}
+        for ele in props:
+            prop_name, _, prop_val = ele.partition("=")
+
+            if not prop_val:
+                system_props[prop_name] = ""
+            else:
+                system_props[prop_name] = prop_val
+
+        return system_props
+
+    def to_option_cmds(self) -> list[str]:
+        """Return the options as a list of strings."""
+        result = self.to_cmd_no_tasks()
+        if self.tasks:
+            for task in self.tasks:
+                result.append(task)
+
+        return result
+
+    def to_cmd_no_tasks(self) -> list[str]:
+        """Return the options only as a list of string.
+
+        Only enabled options are returned.
+
+        Returns
+        -------
+        list[str]
+            The enabled options.
+        """
+        result = []
+
+        if self.help_:
```

Comment:
How would you feel about grouping these append operations into functions based on the categories like with the class parameters? This just seems like a very long single function.